elk/plotting/visualize.py

```diff
@@ -0,0 +1,350 @@
+        df = df[df["eval_dataset"].isin(dataset_names)]
+        df = df[df["train_dataset"].isin(dataset_names)]
+        return df
+
+
+@dataclass
+class Model:
+    df: pd.DataFrame
+    sweep_name: str
+    model_name: str
+    is_transfer: bool
+
+    @classmethod
+    def collect(cls, model_path: Path, sweep_name: str) -> Model:
+        df = pd.DataFrame()
+        model_name = model_path.name
+
+        def handle_csv(dir, eval_dataset, train_dataset):
+            file = dir / "eval.csv"
+            eval_df = pd.read_csv(file)
+            eval_df["eval_dataset"] = eval_dataset
+            eval_df["train_dataset"] = train_dataset
+            return eval_df
+
+        is_transfer = False
+        for train_dir in model_path.iterdir():
+            eval_df = handle_csv(train_dir, train_dir.name, train_dir.name)
+            df = pd.concat([df, eval_df], ignore_index=True)
+            transfer_dir = train_dir / "transfer"
+            if transfer_dir.exists():
+                is_transfer = True
+                for eval_ds_dir in transfer_dir.iterdir():
+                    eval_df = handle_csv(eval_ds_dir, eval_ds_dir.name, train_dir.name)
+                    df = pd.concat([df, eval_df], ignore_index=True)
+
+        df["model_name"] = model_name
+        df["run_name"] = sweep_name
+
+        return cls(df, sweep_name, model_name, is_transfer)
+
+    def render(
+        self,
+        dataset_names: list[str] = ALL_DS_NAMES,
+        score_type="auroc_estimate",
+        ensembling="full",
+    ) -> None:
+        df = self.df
+        model_name = self.model_name
+        sweep_name = self.sweep_name
+        layer_min, layer_max = df["layer"].min(), df["layer"].max()
+        if not (VIZ_PATH / sweep_name).exists():
+            (VIZ_PATH / sweep_name).mkdir()
+        if not (VIZ_PATH / sweep_name / f"{model_name}").exists():
+            (VIZ_PATH / sweep_name / f"{model_name}").mkdir()
+        if self.is_transfer:
+            for layer in range(layer_min, layer_max + 1):
+                filtered = df[(df["layer"] == layer) & (df["ensembling"] == ensembling)]
+                path = VIZ_PATH / sweep_name / f"{model_name}" / f"{layer}.png"
+                if not path.parent.exists():
+                    path.parent.mkdir()
+                fig = TransferEvalHeatmap(
+                    layer, score_type=score_type, ensembling=ensembling
+                ).generate(filtered)
+                fig.write_image(file=path)
+
+        fig = (
+            TransferEvalTrend(dataset_names)
+            .generate(df)
+            .write_image(
+                file=VIZ_PATH / sweep_name / f"{model_name}" / "transfer_eval_trend.png"
+            )
+        )
+
+
+# the following function does too many things.
+# it can be split up into:
+# function that takes a sweep/run and returns a dataframe
+# function that takes df runs / run / model (with or without transfer) and renders it
+
+
+@dataclass
+class Sweep:
```

Comment:
Ok fine, I'll rename it. I thought it'd be fine since it's concise and readable because the two classes are separated by namespaces but chatgpt convinced me it's bad practice